extensions/udapdr/scripts/DSP_Multiple_Reranker.py

```diff
@@ -0,0 +1,704 @@
+
+import os
+import dsp
+from multiprocessing import Process
+import queue
+import torch
+import subprocess as sp
+import openai
+
+from dsp_utilities.generate_synthetic_queries import generate_synthetic_queries
+from dsp_utilities.generate_ColBERTv2_zeroshot_results import generate_ColBERTv2_zeroshot_results
+from dsp_utilities.train_reranker import train_reranker
+from dsp_utilities.evaluate_reranker import evaluate_reranker
+from dsp_utilities.generate_triples import generate_triples
+from dsp_utilities.distill_triples_with_retriever import distill_triples_with_retriever
+from dsp_utilities.evaluate_beir import evaluate_beir
+
+from dsp_utilities.evaluate_lotte_rankings import evaluate_dataset
+from dsp_utilities.pytorchtools import EarlyStopping
+
+import torch.nn as nn
+from transformers import T5Tokenizer, T5EncoderModel, T5ForConditionalGeneration, AutoModelForSeq2SeqLM
+from transformers import BertModel, AutoTokenizer, AutoModel, GPT2Tokenizer
+
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+tqdm.pandas()
+
+import statistics
+import time
+
+import json
+import random
+from random import randrange
+import re
+
+import argparse
```

Comment:
It depends on how detailed we want to be here, but, based on markup from Pycharm, the following imports are not necessary:

> from multiprocessing import Process
> import queue 
> import subprocess as sp
> 
> from dsp_utilities.pytorchtools import EarlyStopping
> 
> import torch.nn as nn
> from transformers import T5Tokenizer, T5EncoderModel, T5ForConditionalGeneration
> from transformers import BertModel,  AutoModel, GPT2Tokenizer
> 
> import statistics
> import time
> 
> import json